internal/pkg/manifest/env.go

```diff
@@ -120,11 +119,11 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID                  *string               `yaml:"id,omitempty"`
-	CIDR                *IPNet                `yaml:"cidr,omitempty"`
-	Subnets             subnetsConfiguration  `yaml:"subnets,omitempty"`
-	SecurityGroupConfig securityGroupConfig   `yaml:"security_group,omitempty"`
-	FlowLogs            VPCFlowLogsArgsorBool `yaml:"flow_logs,omitempty"`
+	ID                  *string                      `yaml:"id,omitempty"`
+	CIDR                *IPNet                       `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration         `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig          `yaml:"security_group,omitempty"`
+	FlowLogs            Union[bool, VPCFlowLogsArgs] `yaml:"flow_logs,omitempty"`
```

Comment:
For `UnmarshalYAML` to work as expected @dannyrandall , do we need to implement the `IsZero` method for custom types like `VPCFLowLogsArgs`?

Reply:
chatted with @KollaAdithya offline, but probably not? reasoning:
If the manifest looks like this:
```
vpc:
  cidr: <config>
```
Then the yaml unmarshaler will never call `FlowLogs.UnmarshalYAML` because it's key isn't present. `FlowLogs` will just be a zero `Union` value, with `IsBasic() == false` and `IsAdvanced() == false`.

However, if it looks like
```
vpc:
  flow_logs:
    random_key: random_value
```
Then `IsAdvanced()` will be **true**, since there's no error unmarshaling that into `VPCFlowLogsArgs`. This is the weird case that you'd need `IsZero()` to handle.

Reply:
actually, thinking about this more - we should probably have `IsZero` to keep `IsAdvanced == false` in that second case.

in general, it'd be nice (i think) if we could have an error if a non-existent struct field is set 🤔

Reply:
ah, i remember why i didn't include that feature https://github.com/go-yaml/yaml/issues/460